homeassistant/components/onvif/switch.py

```diff
@@ -13,6 +15,54 @@
 from .device import ONVIFDevice
 
 
+@dataclass
+class ONVIFSwitchEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    turn_on_fn: Callable[[Any], Any]
```

Comment:
```suggestion
    turn_on_fn: Callable[[ONVIFDevice], Callable[[Any], None]]
```

Reply:
This appears to lead to:
```
switch.py:42: error: Argument "turn_on_fn" to "ONVIFSwitchEntityDescription" has incompatible type "Callable[[ONVIFDevice], Callable[[Profile, Dict[Any, Any]], Coroutine[Any, Any, None]]]"; expected "Callable[[ONVIFDevice], Callable[[Any], None]]"  [arg-type]
switch.py:42: error: Incompatible return value type (got "Callable[[Profile, Dict[Any, Any]], Coroutine[Any, Any, None]]", expected "Callable[[Any], None]")  [return-value]
switch.py:43: error: Argument "turn_off_fn" to "ONVIFSwitchEntityDescription" has incompatible type "Callable[[ONVIFDevice], Callable[[Profile, Dict[Any, Any]], Coroutine[Any, Any, None]]]"; expected "Callable[[ONVIFDevice], Callable[[Any], None]]"  [arg-type]
```
during mypy. I struggled for a long time to get mypy happy on this and ended up on the general one above. Can keep trying.

Reply:
ah nm I think I got it.